src/vos/tests/evt_ctl.c

```diff
@@ -1440,6 +1448,165 @@ test_evt_iter_delete_internal(void **state)
 	assert_int_equal(rc, 0);
 }
 
+static void
+test_evt_variable_record_size_internal(void **state)
+{
+	struct test_arg		*arg = *state;
+	daos_handle_t		 toh;
+	struct evt_entry_in	 entry = {0};
+	int			 rc, count;
+	int			 epoch;
+	const int		val[] = {D_1K_SIZE,
+					D_16K_SIZE};
+	uint64_t		data_size;
+	char                     *data;
+
+	rc = evt_create(ts_feats, 13, arg->ta_uma, arg->ta_root,
+		DAOS_HDL_INVAL, &toh);
+	assert_int_equal(rc, 0);
+	for (count = 0; count < sizeof(val)/sizeof(int); count++) {
+		/* Try to insert a bunch of entries with variable data sizes */
+		data_size = val[count];
+		D_ALLOC(data, data_size);
+		strcpy(data, "EVTree: Test Variable Record Size");
+		for (epoch = 1; epoch < 3 ; epoch++) {
+			entry.ei_rect.rc_ex.ex_lo = epoch;
+			entry.ei_rect.rc_ex.ex_hi = epoch + data_size;
+			entry.ei_rect.rc_epc = epoch;
+			memset(&entry.ei_csum, 0, sizeof(entry.ei_csum));
+			entry.ei_ver = 0;
+			entry.ei_inob = data_size;
+			rc = bio_alloc_init(arg->ta_utx, &entry.ei_addr,
+					    data, data_size);
+			assert_int_equal(rc, 0);
+			rc = evt_insert(toh, &entry);
+			if (count > 0)
+				assert_int_not_equal(rc, 0);
+			else
+				assert_int_equal(rc, 0);
+		}
+	}
+	rc = evt_destroy(toh);
+	assert_int_equal(rc, 0);
+}
+
+static void
+test_evt_various_data_size_internal(void **state)
+{
+	struct test_arg		*arg = *state;
+	daos_handle_t		 toh;
+	daos_handle_t		 ih;
+	struct evt_entry_in	 entry = {0};
+	int			 rc, count;
+	int			 epoch;
+	const int		val[] = {D_1K_SIZE,
+					D_16K_SIZE,
+					D_256K_SIZE,
+					D_512K_SIZE,
+					D_1M_SIZE,
+					D_256M_SIZE};
+	uint64_t		data_size;
+	char                     *data;
+	struct evt_entry_array	 ent_array;
+	struct evt_entry	 *ent;
+	bio_addr_t		 addr;
+	struct evt_extent	 extent;
+	daos_epoch_range_t	 epr;
+
+	for (count = 0; count < sizeof(val)/sizeof(int); count++) {
+		rc = evt_create(ts_feats, 13, arg->ta_uma, arg->ta_root,
+			DAOS_HDL_INVAL, &toh);
+		assert_int_equal(rc, 0);
+		rc = utest_sync_mem_status(arg->ta_utx);
+		assert_int_equal(rc, 0);
+		data_size = val[count];
+		D_PRINT("Data Size: %ld\n", data_size);
+		D_ALLOC(data, data_size);
+		strcpy(data, "EVTree: Out of Memory");
+		epr.epr_lo = 0;
+		/* Loop does the following : evt_insert,
+		* evt_find (first epoch) and evt_delete (random deletes)
+		* till out of space condition
+		*/
+		for (epoch = 1; ; epoch++) {
+			entry.ei_rect.rc_ex.ex_lo = epoch;
+			entry.ei_rect.rc_ex.ex_hi = epoch + data_size;
+			entry.ei_rect.rc_epc = epoch;
+			memset(&entry.ei_csum, 0, sizeof(entry.ei_csum));
+			entry.ei_ver = 0;
+			entry.ei_inob = data_size;
+			rc = bio_alloc_init(arg->ta_utx, &entry.ei_addr,
+					    data, data_size);
+			if (rc != 0)
```

Comment:
may want to assert it's a proper error code

Reply:
Fixed.